batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/controller/StudentController.java

```diff
@@ -0,0 +1,214 @@
+            req.setAttribute("roll", rollText);
+            create(req, resp);
+        } catch (NumberFormatException e){
+            req.setAttribute("errorMessage", "number format error in 'roll'");
+            req.setAttribute("name", name);
+            req.setAttribute("address", address);
+            req.setAttribute("dob", dob);
+            req.setAttribute("department", department);
+            req.setAttribute("batch", batch);
+            req.setAttribute("roll", rollText);
+            create(req, resp);
+        }
+
+    }
+
+
+    private void edit(HttpServletRequest req, HttpServletResponse resp, Integer id) throws ServletException, IOException {
+
+        try {
+            Student student = studentService.fetchById(id);
+            req.setAttribute("student",student);
+            req.getServletContext().getRequestDispatcher("/WEB-INF/views/editStudent.jsp").forward(req, resp);
+        } catch (DataException e) {
+            e.printStackTrace();
+            resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            req.setAttribute("errorMessage", e.getMessage());
+            req.getServletContext().getRequestDispatcher("/WEB-INF/views/error.jsp").forward(req, resp);
+        }
+    }
+
+
+    private void editProcess(HttpServletRequest req, HttpServletResponse resp, Integer id) throws ServletException, IOException {
+        String name= req.getParameter("name");
+        String address = req.getParameter("address");
+        String dob = req.getParameter("dob");
+        DateFormat dateFormat = new SimpleDateFormat("yyyy-mm-dd", Locale.ENGLISH);
+        Date date=null;
+        try {
+            date = dateFormat.parse(dob);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+
+        Integer roll = Integer.parseInt(req.getParameter("roll"));
+        String department = req.getParameter("department");
+        String batch = req.getParameter("batch");
+
+        Student student = new Student();
+        student.setId(id);
+        student.setBatch(batch);
+        student.setDepartment(department);
+        student.setRoll(roll);
+
+        student.setName(name);
+        student.setAddress(address);
+        student.setDob(date);
+
+        try {
+            studentService.update(student);
+            resp.sendRedirect("/students");
+        } catch (DataException e) {
+            e.printStackTrace();
+            resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            req.setAttribute("errorMessage", e.getMessage());
+            req.getServletContext().getRequestDispatcher("/WEB-INF/views/error.jsp").forward(req, resp);
+        }
+    }
+
+
+    private void deleteProcess(HttpServletRequest req, HttpServletResponse resp, int id) throws ServletException, IOException {
+        try {
+            studentService.delete(id);
+            resp.sendRedirect("/students");
+        } catch (DataException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)